src/main/java/Storage.java

```diff
@@ -0,0 +1,82 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    private String filePath = "";
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> outputList = new ArrayList<>();
+        try {
+            File f = new File(filePath);
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String savedTask = s.nextLine();
+                String[] savedTaskSplit = savedTask.split("\\|");
+                if (savedTaskSplit[0].equals("T")) {
```

Comment:
Perhaps you could use switch-case statements instead of if-else loops here to make the different scenarios clearer to the reader!

Reply:
Yes, good idea since there are many cases to consider. I used if-else blocks due to personal preference, hope the readability was still fine for you!